salt/utils/win_functions.py

```diff
@@ -152,8 +152,10 @@ def get_sam_name(username):
     Everything is returned lower case
 
     i.e. salt.utils.fix_local_user('Administrator') would return 'computername\administrator'
+
+    .. note:: Long computer names are truncated to 15 characters
     '''
     if '\\' not in username:
-        username = '{0}\\{1}'.format(platform.node(), username)
+        username = '{0}\\{1}'.format(platform.node()[:15], username)
```

Comment:
Is this change related to the `gid=gid` fix? I don't see an explanation of why this part is included in this PR or why this would be necessary.

If it should be included here, can you add a some documentation or an explanation of why this is truncated? That will help the will code maintenance in the future.

Reply:
Can you also explain why truncated this is needed? This seems like it would really potentially cause a lot of confusion and potential duplication for users of this function.

Reply:
@rallytime Yeah, this is also a part of the fix. When I originally "Fixed" this module, I was on a system with a shorter hostname, so I didn't see the issue with longer hostnames.

@cachedout The problem is that in the execution module when it lists the members of a group it lists them as `hostname\\groupname`. The hostname is truncated to 15 characters. So when the state tries to figure out if the user is already in the group it thinks it's not because it's looking for the `fullhostname\\username` instead of the truncated one. The `win_function.get_sam_name` was written specifically for the group thing on Windows.

Reply:
Windows does not permit computer names that exceed 15 characters.  The maximum length of the host name and of the fully qualified domain name (FQDN) is 63 bytes per label and 255 characters per FQDN. The latter is based on the maximum path length possible with an Active Directory Domain name with the paths needed in SYSVOL, and this needs to obey to the 260 character MAX_PATH limitation.  https://support.microsoft.com/en-au/help/909264/naming-conventions-in-active-directory-for-computers-domains-sites-and

Not sure why name would be allowed, unless then name being returned has a domain in it. (which maybe it needs to be stripped off)  You can have long usernames but they must be in the format of <username>@<realm>  where realm is <DOMAIN>.<rest of DNS Domain>

Reply:
win32security.TranslateName might be usefull

Reply:
`win32security.TranslateName` seems to only work when the host is joined to a domain

Reply:
`sid = win32security.GetBinarySid(uid)` # This is not correct as it excepts  SID not a UserName but may give you a glue.
`name, domain, account_type = win32security.LookupAccountSid(None, sid)`

Reply:
Backwards and then Forward lookup
```
sid_str = win32security.LookupAccountName(None, username)[0]
sid = win32security.GetBinarySid(sid_str)
name, domain, account_type = win32security.LookupAccountSid(None, sid)
```
?

Reply:
That might work...